fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -2831,4 +2856,47 @@ public static boolean isCompartmentSearch(FHIRSearchContext searchContext) {
         return result;
     }
 
+    /**
+     * Determine if a search parameter expression references an element of type Canonical.
+     * @param resourceType
+     * @param expression
+     * @return
+     */
+    private static boolean isCanonicalSearchParm(Class<?> resourceType, String expression) {
+        // Parse the expression to determine element type it refers to.
+        // First split by union operator.
+        for (String subExpression : expression.split("\\|")) {
+            Class<?> dataType = resourceType;
+            
+            // Strip any enclosing parens
+            subExpression = subExpression.trim();
+            if (subExpression.startsWith("(") && subExpression.endsWith(")")) {
+                subExpression = subExpression.substring(1, subExpression.length()-1);
+            }
+            
+            // Strip out '.where()' functions, handling nested parentheses
+            subExpression = subExpression
+                    .replaceAll("\\.where(?=\\()(?:(?=.*?\\((?!.*?\\1)(.*\\)(?!.*\\2).*))(?=.*?\\)(?!.*?\\2)(.*)).)+?.*?(?=\\1)[^(]*(?=\\2$)", "");
+            
+            // Split into components
+            for (String component : subExpression.split("\\.")) {
+                component = component.trim();
+                if ((dataType == null || !component.equals(dataType.getSimpleName())) && !component.startsWith("where")) {
+                    if (component.contains(" as ")) {
+                        dataType = ModelSupport.getDataType(component.substring(component.indexOf(" as ") + 4));
+                    } else if (component.startsWith("as(")) {
+                        dataType = ModelSupport.getDataType(component.substring(3, component.indexOf(")")).trim());
+                    } else if (ModelSupport.isResourceType(component)) {
+                        dataType = ModelSupport.getResourceType(component);
+                    } else {
+                        dataType = ModelSupport.getElementType(dataType, component);
+                    }
+                }
+            }
+            if ("Canonical".equals(dataType.getSimpleName())) {
```

Comment:
Feels like "Canonical" should be a constant in SearchConstants.java.

Reply:
I used Canonical.class.getSimpleType() instead.